File: top-interview-150/58-length-of-last-word.py
'''
* 58. Length of Last Word
* https://leetcode.com/problems/length-of-last-word/description/?envType=study-plan-v2&envId=top-interview-150
* Time: O(n), Space: O(1)
* Difficulty: Easy
'''

#Esta es la solución larga. Se puede hacerf con un s.strip().split() y del tirón. Eso está hecho en la solución con Java
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        i = len(s) - 1
        count = 0

        while(i >= 0 and s[i] == " "):
            i -= 1

        while(i >= 0 and s[i] != " "):
            count += 1
            i -= 1


        return count

File: top-interview-150/58-length-of-last-word.java
/**
 * 58. Length of Last Word
 * https://leetcode.com/problems/length-of-last-word/description/?envType=study-plan-v2&envId=top-interview-150
 * Time: O(n), Space: O(n). La solución de O(1) de espacio es la de Python.
 * Difficulty: Easy
 */


class Solution {
    public int lengthOfLastWord(String s) {


        String[] splits = s.trim().split(" ");

        return splits[splits.length-1].length();
    }
}
